editoast/src/views/study.rs

```diff
@@ -124,29 +143,44 @@ async fn create(
     db_pool: Data<DbPool>,
     data: Json<StudyCreateForm>,
     project: Path<i64>,
-) -> Result<Json<StudyWithScenarios>> {
+) -> Result<Json<StudyResponse>> {
     let project_id = project.into_inner();
-    // Check if project exists
-    let project = match Project::retrieve(db_pool.clone(), project_id).await? {
-        None => return Err(ProjectError::NotFound { project_id }.into()),
-        Some(project) => project,
-    };
 
-    // Create study
-    let study: Study = data.into_inner().into_study(project_id)?;
-    let study = study.create(db_pool.clone()).await?;
+    let mut conn = db_pool.get().await?;
 
-    // Update project last_modification field
-    let project = project.update_last_modified(db_pool).await?;
-    project.expect("Project should exist");
+    let (study, project) = conn
+        .transaction::<_, InternalError, _>(|conn| {
+            async {
+                // Check if project exists
+                let Some(project) = Project::retrieve(db_pool.clone(), project_id).await? else {
+                    return Err(ProjectError::NotFound { project_id }.into());
+                };
+
+                // Create study
+                let _study: Study = data.into_inner().into_study(project_id)?;
+                let _study = _study.create(db_pool.clone()).await?;
```

Comment:
Likewise about db_pool. Also, it's a detail, but `_variable_name` is by convention how you name an unused variable. 
```suggestion
                let study = data.into_inner().into_study(project_id)?.create_conn(conn).await?;
```